httpcore/_types.py

```diff
@@ -9,3 +9,4 @@
 URL = Tuple[bytes, bytes, Optional[int], bytes]
 Headers = List[Tuple[bytes, bytes]]
 TimeoutDict = Dict[str, Optional[float]]
+SocketAddress = Tuple[Union[str, bytes, bytearray], int]
```

Comment:
Would it be reasonable to keep this as tightly constrained as possible?
Presumably `Tuple[bytes, int]` is sufficient right?

Reply:
If source-port is removed, this type isn't needed at all, as the address could just be a `bytes`.

Reply:
Right, yup.